libs/experimental/langchain_experimental/llms/ollama_functions.py

```diff
@@ -74,18 +82,32 @@ def _is_pydantic_class(obj: Any) -> bool:
     )
 
 
+def _is_pydantic_object(obj: Any) -> bool:
+    return isinstance(obj, BaseModel)
+
+
 def convert_to_ollama_tool(tool: Any) -> Dict:
     """Convert a tool to an Ollama tool."""
+    description = None
     if _is_pydantic_class(tool):
         schema = tool.construct().schema()
-        definition = {"name": schema["title"], "properties": schema["properties"]}
-        if "required" in schema:
-            definition["required"] = schema["required"]
+        name = schema["title"]
+    elif _is_pydantic_object(tool):
+        schema = tool.get_input_schema().schema()
+        name = tool.get_name()
+        description = tool.description
```

Comment:
have you tried with subclass of BaseTool ? I am not able to make it work and it fails here

Reply:
@leobenkel 
Can you share your code so I can test it to identify the root cause?
The code checks for pydantic BaseModel which should cover the BaseTool subclass as well but I want to make sure.

Reply:
I am not allowed to share this code sadly. But I made a simple tool using a `class MyTool(BaseTool)` and that did not work when trying to get the `schema()` on line 96

Reply:
I see that in your notebook, you are using the `@tool` , could you try with the `class ...(BaseTool)` ?

Reply:
Sure. I'll give this a try and share an update.

Reply:
thank you